browser_patches/webkit/patches/bootstrap.diff

```diff
@@ -12362,20 +12369,25 @@ index c61bb3fd2ee0..0fee368e9364 100644
 +    , m_uuid(createCanonicalUUIDString())
  {
 +    if (auto* instrumentation = m_dataStore->downloadInstrumentation())
-+      instrumentation->downloadCreated(m_uuid, m_request, frameInfoData, originatingPage);
++      instrumentation->downloadCreated(m_uuid, m_request, frameInfoData, originatingPage, this);
  }
  
  DownloadProxy::~DownloadProxy()
-@@ -79,6 +84,8 @@ void DownloadProxy::cancel(CompletionHandler<void(API::Data*)>&& completionHandl
+@@ -75,9 +80,12 @@ static RefPtr<API::Data> createData(const IPC::DataReference& data)
+ void DownloadProxy::cancel(CompletionHandler<void(API::Data*)>&& completionHandler)
+ {
+     if (m_dataStore) {
+-        m_dataStore->networkProcess().sendWithAsyncReply(Messages::NetworkProcess::CancelDownload(m_downloadID), [this, protectedThis = makeRef(*this), completionHandler = WTFMove(completionHandler)] (const IPC::DataReference& resumeData) mutable {
++        auto* instrumentation = m_dataStore->downloadInstrumentation();
```

Comment:
Is there a reason the instrumentation cannot be accessed inside the callback as before?

Reply:
Inside downloadMapProxy.downloadFinished the m_dataStore reference gets cleared which leads to a accessing a null pointer otherwise.